Testing Your Ruby Code - Enumerable

bmuhamood opened the pull request:
- Tested linters and all working fine. 
- All provided Test cases working fine. 
- Follow Git hub flow.

Commit:
added all rspec files

Commit:
added required directories for rspec

Commit:
added all the test cases

Commit:
corrected 2 method errors

Commit:
fixed my each and my_each_with_index methods

Commit:
removed .DS file

Commit:
orgised lib file

Commit:
added linters, Readme, and also Fixed all method errors

Commit:
corrected grammery errors in Readme

Commit:
fixing linters errors

Commit:
fixing linter errors

Commit:
fixing linters errors

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters erros

Commit:
fixing linters erros

Commit:
Fixed more linters errors

Commit:
fixing more linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
linters

Commit:
linters

Commit:
fixing linters

Commit:
fixing linters errors

Commit:
added Gem files

Commit:
added Gem files

Commit:
added Gem files

Commit:
added Gem files

Commit:
linters

Commit:
linters

Commit:
linters

Commit:
linters

Commit:
linters

Commit:
linters

Commit:
linters

Commit:
linters

Commit:
fixing all linters errors

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
linters

Commit:
linters

Commit:
linters

expjazz commented on spec/enumerables_spec.rb:
```diff
@@ -0,0 +1,196 @@
+
+describe '#my_each' do
+  it 'return enumerator if block is not given' do
+    arr = [1, 2, 3, 4]
```
- [ ] You are defining this array a lot of times. For variables like this, you can use the _:let_  to define them one time at the top of the tests. This will make your code cleaner. Do that for all variables you are using multiple times.

expjazz reviewed (changes requested):
## Status: Changes Required :recycle: 

Hey team :+1: 

You did a very good job here. Your tests are covering a lot of edge cases, and they are very well written. 
 However, we still have a couple of details to fix:

- [ ]  You have disabled the LineLength linter. I think that is not necessary. Try to fix this issue without disabling it.

- [ ] Use the _let_ keyword to define variables at the top to not repeat yourself (for this, check the inline comment below for more detail).

Other than that it is all good. Marvelous job here :1st_place_medal: 

### Please make the required changes and resubmit for a new code review once you are ready
Good Luck and Happy Coding :pray:

_P.S. If you have any questions about my review or need help, you can message me on slack (@ Expedito Andrade).:smile:_

Commit:
reduced lines

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixed all linters errors

Commit:
fixing linters errors

Commit:
broken my_none which was too long

Commit:
broken my_none aurgument

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixed all linters errors

Commit:
fixed all linters errors

Commit:
linters

Commit:
created let variable as arr and used later to avoid DRY

Commit:
fixing linters errors

Commit:
fixing linters

Commit:
fixing linters

Commit:
correct empty line issues in linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

Commit:
fixing linters

bmuhamood commented:
Thanks a lot for the best review. 

- Enabled the LineLength linter and fixed manually. 
- Removed all duplicate Variables and created one i.e. **let** for using later so as to make code more cleaner. 
- Fixed all linters errors. 
- All test are working fine.

MuriloRoque reviewed (approved):
# Status: Approved :green_circle: 

### Hello @bmuhamood :wave: 

### Thank you for making all of the required changes, your project is approved :clap: :clap: :clap: 

### You may merge, good luck with your next projects :rocket: 

Murilo Roque Paiva da Silva

- Github: [@muriloroque](https://github.com/MuriloRoque)
- Linkedin: [MuriloRoque](https://www.linkedin.com/in/murilo-roque-b1268741/)

[![Twitter Follow](https://img.shields.io/twitter/follow/MuriloRoquePai1?label=Follow%20Murilo%20on%20Twitter&style=social)](https://twitter.com/MuriloRoquePai1)
